frontend/widgets/AudioMixer.cpp

```diff
@@ -0,0 +1,1035 @@
+/******************************************************************************
+    Copyright (C) 2025 by Taylor Giampaolo <warchamp7@obsproject.com>
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+#include <widgets/AudioMixer.hpp>
+#include <components/MenuCheckBox.hpp>
+
+#include <QAction>
+#include <QCheckBox>
+#include <QWidgetAction>
+
+#include <widgets/OBSBasic.hpp>
+#include <dialogs/NameDialog.hpp>
+#include <utility/item-widget-helpers.hpp>
+
+#include "moc_AudioMixer.cpp"
+
+namespace {
+bool isHideInMixer(obs_source_t *source)
+{
+	OBSDataAutoRelease priv_settings = obs_source_get_private_settings(source);
+	bool hidden = obs_data_get_bool(priv_settings, "mixer_hidden");
+
+	return hidden;
+}
+
+bool isPinnedInMixer(obs_source_t *source)
+{
+	OBSDataAutoRelease priv_settings = obs_source_get_private_settings(source);
+	bool hidden = obs_data_get_bool(priv_settings, "mixer_pinned");
+
+	return hidden;
+}
+
+bool isSourceAudioActive(obs_source_t *source)
+{
+	bool active = obs_source_active(source) && obs_source_audio_active(source);
+
+	return active;
+}
+
+bool isVolumeLocked(obs_source_t *source)
+{
+	OBSDataAutoRelease priv_settings = obs_source_get_private_settings(source);
+	bool lock = obs_data_get_bool(priv_settings, "volume_locked");
+
+	return lock;
+}
+
+} // namespace
+
+AudioMixer::AudioMixer(QWidget *parent) : QFrame(parent)
+{
+	showInactive = config_get_bool(App()->GetUserConfig(), "BasicWindow", "MixerShowInactive");
+	keepInactiveRight = config_get_bool(App()->GetUserConfig(), "BasicWindow", "MixerKeepInactiveRight");
```

Comment:
just my personal taste, but I prefer this to default to true

Reply:
I'm trying to avoid any major changes to behaviour for folks, but these options being exposed so easily now means it may not be bad to improve the discoverability here.